### PR TITLE
Calypsoify: add new post link to calypsoifyGutenberg

### PR DESCRIPTION
**Pull request by creativecoder**
Companion PR for https://github.com/Automattic/wp-calypso/pull/52122

#### Changes proposed in this Pull Request:

Adds the link for a site to create a new post in the Calypsoified editor.

This is used for overriding the link when the editor is Calypsoified, so that creating a new post is also Calypsoified, rather than opened in wp-admin.

#### Jetpack product discussion

n/a

#### Does this pull request change what data or activity we track or use?

no

#### Testing instructions:

* Open a post in a Jetpack connected site on WordPress.com in the calypsofied editor.
* Search the source of the page to find `createNewPostUrl` in a js object output on the page, with the correct WP.com editor url for the site (`https://wordpress.com/post/{site}`)

### CODE REVIEW

File: projects/plugins/jetpack/modules/calypsoify/class-jetpack-calypsoify.php [hunk 1]
```diff
@@ -87,7 +88,8 @@ public function enqueue_for_gutenberg() {
 			'calypsoifyGutenberg',
 			array(
 				'closeUrl'                => $this->get_close_gutenberg_url(),
-				'manageReusableBlocksUrl' => $this->get_calypso_origin() . '/types/wp_block/' . ( new Status() )->get_site_suffix(),
+				'manageReusableBlocksUrl' => $this->get_calypso_origin() . '/types/wp_block/' . $site_suffix,
+				'createNewPostUrl'        => $this->get_calypso_origin() . '/post' . $site_suffix,
```
Comment (jeherve) [inline projects/plugins/jetpack/modules/calypsoify/class-jetpack-calypsoify.php:92]: Shouldn't there be an additional `/` here?

```suggestion
				'createNewPostUrl'        => $this->get_calypso_origin() . '/post/' . $site_suffix,
```

File: changelog/add-calypsoify-create-new-post-url [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Significance: patch
```
Comment (jeherve) [inline changelog/add-calypsoify-create-new-post-url:1]: It looks like you added the changelog file to the root of the monorepo. Could you remove it, and try something like this instead? That may make things easier.

https://user-images.githubusercontent.com/426388/117431088-1bcef500-af29-11eb-82fd-c89d917dc4f2.mov

**Review by jeherve — CHANGES_REQUESTED** [review]

File: projects/plugins/jetpack/modules/calypsoify/class-jetpack-calypsoify.php (hunk 1 above)
Comment (creativecoder) [reply projects/plugins/jetpack/modules/calypsoify/class-jetpack-calypsoify.php]: 🤦‍♂️ thanks for catching that, now updated

File: changelog/add-calypsoify-create-new-post-url (hunk 2 above)
Comment (creativecoder) [reply changelog/add-calypsoify-create-new-post-url]: Done. Thanks for the tip!

**Pull request merged**